lib/include/jxl/decode.h

```diff
@@ -236,6 +236,33 @@ typedef enum {
   JXL_DEC_JPEG_RECONSTRUCTION = 0x2000,
 } JxlDecoderStatus;
 
+/** Rewinds decoder to the beginning. The same input must be given again from
+ * the beginning of the file and the decoder will emit events from the beginning
+ * again. When rewinding (as opposed to JxlDecoderReset), the decoder keeps
+ * state about the image, which it can use to skip to a requested frame more
+ * efficiently with JxlDecoderSkipFrames. After rewind,
+ * JxlDecoderSubscribeEvents must be used again, and it is feasible to leave out
+ * events that were already handled before, such as JXL_DEC_BASIC_INFO and
+ * JXL_DEC_COLOR_ENCODING, since they will provide the same information as
+ * before.
+ * @param dec decoder object
+ */
+JXL_EXPORT void JxlDecoderRewind(const JxlDecoder* dec);
+
+/** Makes the decoder skip the next `amount` frames. If this is used after
+ * JxlDecoderRewind to skip frames that were already decoded before, the decoder
+ * is able to skip more efficiently, as it can skip decoding earlier frames for
+ * which it knows the later frames don't depend on them. A frame here is defined
```

Comment:
... earlier frames that it knows later frames don't depend on?

Reply:
Yes, that's better, done